File: webui/src/main/java/com/taskadapter/webui/LastVersionLoader.java
package com.taskadapter.webui;

import com.taskadapter.http.HttpCaller;
import com.taskadapter.webui.service.TaPropertiesLoader;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class LastVersionLoader {
    private static final Logger log = LoggerFactory.getLogger("default");

    /**
     * check the last TA version available for download on the website.
     */
    public static String loadLastVersion() {
        var url = TaPropertiesLoader.getUpdateAppUrl();
        try {
            var lastVersionString = HttpCaller.getAsString(url);
            return lastVersionString.trim();
        } catch (Exception e) {
            log.error("Cannot load last version info: " + e.toString());
            return "unknown";
        }
    }
}

File: webui/src/main/scala/com/taskadapter/webui/LastVersionLoader.scala
package com.taskadapter.webui

import com.taskadapter.connector.PropertiesUtf8Loader
import com.taskadapter.http.HttpCaller
import org.slf4j.LoggerFactory

object LastVersionLoader {
  private val log = LoggerFactory.getLogger("default")

  /**
    * check the last TA version available for download on the website.
    */
  def loadLastVersion: String = {
    val properties = PropertiesUtf8Loader.load("taskadapter.properties")
    val url = properties.getProperty("update_site_url")
    try {
      val lastVersionString = HttpCaller.get(url, classOf[String])
      lastVersionString.trim
    } catch {
      case e: Exception => log.error("Cannot load last version info: " + e.toString)
        "unknown"
    }
  }
}